<?xml version="1.0"?>

<launch>
    <!-- Set up hardware configurations and start up nodes related to hardware components. -->
    <group ns="hardware">
        <rosparam command="load" file="$(find expansion_board_driver)/config/hardware.yaml"/>
    </group>
    <node name="motor_controller_node"          pkg="expansion_board_driver" type="motor_controller.py"/>
    <node name="fan_controller_node"            pkg="expansion_board_driver" type="fan_controller.py"/>
    <node name="led_controller_node"            pkg="expansion_board_driver" type="led_controller.py"/>
    <node name="rear_collision_detector_node"   pkg="expansion_board_driver" type="rear_collision_detector.py"/>

    <!-- Set up RPLiDAR-->
    <node   name="map_to_laser_broadcaster"     type="static_transform_publisher"       pkg="tf"    args="0 0 0 0 0 0 1 map laser_frame 10"/>
    <node   name="rplidarNode"                  type="rplidarNode"                      pkg="rplidar_ros"/>
    <param  name="serial_port"                  type="string"   value="/dev/ttyUSB0"/>
    <param  name="serial_baudrate"              type="int"      value="115200"/>
    <param  name="frame_id"                     type="string"   value="laser"/>
    <param  name="inverted"                     type="bool"     value="false"/>
    <param  name="angle_compensate"             type="bool"     value="true"/>
</launch>